Solutions/Microsoft 365 Defender/Analytic Rules/SUNSPOTHashes.yaml

```diff
@@ -31,10 +31,13 @@ query: |
   (DeviceImageLoadEvents
   | where InitiatingProcessSHA256 in (SUNSPOT_Hashes))
   | extend HostCustomEntity = DeviceName, timestamp=TimeGenerated
+  | extend HostName = tostring(split(DeviceName, '.', 0)[0]), DnsDomain = tostring(strcat_array(array_slice(split(DeviceName, '.'), 1, -1), '.'))
```

Comment:
"HostCustomEntity = DeviceName" This is not required, so you can remove it.

Better You can combine these two lines like,
extend timestamp=TimeGenerated, HostName = tostring(split(DeviceName, '.', 0)[0]), DnsDomain = tostring(strcat_array(array_slice(split(DeviceName, '.'), 1, -1), '.'))

Reply:
done